### PR TITLE
Cookie banner - update content and functionality

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?

- Update cookie banner to follow updates from gov.uk and nhs.uk

Issue - #1913 

### What are the breaking changes?
The **breaking changes** are updates to a number of parameters to configure the new version of the component:

`primaryButtonText` is now `acceptButtonText`
`secondaryButtonText` is now `rejectButtonText`
`settingsLinkText` and `settingsLinkTextURL` are new parameters

### How to review

- Check the new content and functionality (when you click either accept/reject you'll need to clear the cookie `ons_message_seen` and reload)
- Test are passing

### CODE REVIEW

**Comment by jrbarnes9** [comment]
I think the banner content could do with being re-worded to sound less alarming, but will leave that to content designers...
- "small file that we put on your device." – Think this needs more explanation. Maybe avoid the "we"? Could we say something like "cookies temporarily store information for your browser to..."?
- "Some cookies are essential" –> e.g. "We need to stored some essential cookies to make the site work"

File: src/components/cookies-banner/examples/cookies-banner/index.njk [hunk 1]
```diff
@@ -4,9 +4,11 @@
 {% from "components/cookies-banner/_macro.njk" import onsCookiesBanner %}
 {{
     onsCookiesBanner({
-        "statementTitle": 'Tell us whether you accept cookies',
-        "statementText": 'We use <a href="#0">cookies to collect information</a> about how you use census.gov.uk. We use this information to make the website work as well as possible and improve our services.',
-        "confirmationText": 'You’ve accepted all cookies. You can <a href="#0">change your cookie preferences</a> at any time.',
-        "secondaryButtonUrl": '#'
+        "statementTitle": 'Cookies on census.gov.uk',
```
Comment (jrbarnes9) [inline src/components/cookies-banner/examples/cookies-banner/index.njk:7]:
```suggestion
        "statementTitle": 'Cookies on ons.gov.uk',
```

**Comment by jrbarnes9** [comment]
Think we could probably improve the layout of the controls at smaller breakpoints. Don't think we necessarily need the buttons in the grid col, they can probably always fill the 12 col to stay inline, then perhaps all stack vertically below medium?

Screenshot 2022-02-28 at 13 50 13
Screenshot 2022-02-28 at 13 47 09

**Comment by jrbarnes9** [comment]
Content from DP content designers...

> ## Cookies on ons.gov.uk
>
> Cookies are small files stored on your device when you visit a website. We use some essential cookies to make this website work.
>
> We would like to set [additional cookies]() to remember your settings and understand how you use the site. This helps us to improve our services.
>
> [ Accept additional cookies ] 
> [ Reject additional cookies ]
> [View cookies]()


Screenshot 2022-06-30 at 14 14 29

**Comment by jrbarnes9** [comment]
Welsh content for banner:

> ## Cwcis ar ons.gov.uk
>
> Ffeiliau bach a gaiff eu storio ar eich dyfais pan fyddwch yn mynd ar wefan yw cwcis.
> Rydym ni'n defnyddio rhai cwcis hanfodol i wneud i'r wefan hon weithio. 
>
> Hoffem osod [cwcis ychwanegol](#0) er mwyn cofio eich gosodiadau a deall sut rydych 
> chi'n defnyddio'r wefan. Mae hyn yn ein helpu ni i wella ein gwasanaethau. 
>
> [ Derbyn cwcis ychwanegol ]
> [ Gwrthod cwcis ychwanegol ]
> [Gweld cwcis](#0)

**Comment by boxadesign** [comment]
@jrbarnes9 

Also need the new confirmation message translated:

You’ve accepted all additional cookies.You can [change your cookie preferences](#0) at any time.
You’ve rejected all additional cookies.You can [change your cookie preferences](#0) at any time.

**Review by kruncher — APPROVED** [review]

**Comment by jrbarnes9** [comment]
> @jrbarnes9
> 
> Also need the new confirmation message translated:
> 
> You’ve accepted all additional cookies.You can [change your cookie preferences](#0) at any time. You’ve rejected all additional cookies.You can [change your cookie preferences](#0) at any time.

I've requested this from DP content designer.

**Comment by jrbarnes9** [comment]
New content for confirmation banner, and welsh translation:

## Accepted cookies confirmation

English
> You have accepted all additional cookies. You [can change your cookie preferences](/cookies) 
at any time.

Welsh
> Rydych chi wedi derbyn yr holl gwcis ychwanegol. Gallwch chi [newid eich 
dewisiadau o ran cwcis](/cwcis) ar unrhyw adeg.

## Rejected cookies confirmation
English
> You have rejected all additional cookies. You can [change your cookie preferences](/cookies) at
any time.

Welsh
> Rydych chi wedi gwrthod yr holl gwcis ychwanegol. Gallwch chi [newid eich 
dewisiadau o ran cwcis](/cwcis) ar unrhyw adeg.

File: src/components/cookies-banner/_macro-options.md [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
+| serviceName                | string | false    | The name of the service/website that follows the `statementTitle` (default is 'ons.gov.uk')                                                                                                                                                                                                                                                                                     |
+| statementText              | string | false    | The text content for the cookie banner. HTML is allowed to create paragraphs (default is 'Cookies are small files stored on your device when you visit a website. We store some cookies that are essential to make the site work. We would like to set additional cookiesto remember your settings and understand how you use the site. This helps us to improve our services') |
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro-options.md:7]:
```suggestion
| statementText              | string | false    | The text content for the cookie banner. HTML is allowed to create paragraphs. Default is: ”Cookies are small files stored on your device when you visit a website. We use some essential cookies to make this website work. We would like to set additional cookies to remember your settings and understand how you use the site. This helps us to improve our services.” |
```

File: src/patterns/help-users-to/cookies-settings/examples/cookie-settings-page/index.njk [hunk 3]
```diff
@@ -45,10 +45,12 @@ height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
...
-        "statementTitle": 'Tell us whether you accept cookies',
-        "statementText": 'We use <a href="#0">cookies to collect information</a> about how you use census.gov.uk. We use this information to make the website work as well as possible and improve our services.',
-        "confirmationText": 'You’ve accepted all cookies. You can <a href="#0">change your cookie preferences</a> at any time.',
-        "secondaryButtonUrl": '#'
+        "statementTitle": 'Cookies on ons.gov.uk',
+        "statementText": '<p>Cookies are small files stored on your device when you visit a website. We store some cookies that are essential to make the site work.<p>
+                          <p>You can allow additional cookies that remember your settings and help us improve our services by understanding how you use the site.</p>',
+        "acceptedText": 'You’ve accepted all additional cookies.',
+        "rejectedText": 'You’ve rejected all additional cookies.',
+        "preferencesText": 'You can <a href="#0">change your cookie preferences</a> at any time.'
```
Comment (jrbarnes9) [inline src/patterns/help-users-to/cookies-settings/examples/cookie-settings-page/index.njk:53]: Content needs updating

File: src/components/cookies-banner/_macro.njk [hunk 4]
```diff
@@ -1,40 +1,84 @@
...
+        {% set acceptedText = 'Rydych chi wedi derbyn yr holl gwcis ychwanegol.' %}
+        {% set rejectedText =  'Rydych chi wedi gwrthod yr holl gwcis ychwanegol. ' %}
+        {% set preferencesText =  'Gallwch chi <a href="' + settingsLinkURL + '">newid eich dewisiadau o ran cwcis</a> ar unrhyw adeg.' %}
+        {% set confirmationButtonText = 'Cuddio' %}
+        {% set confirmationButtonTextAria = 'neges hon' %}
+    {% else %}
+        {% set ariaLabel = 'Cookies banner' %}
+        {% set serviceName = 'ons.gov.uk' %}
+        {% set statementTitle = 'Cookies on' %}
+        {% set settingsLinkText = 'Cookie settings' %}
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro.njk:20]: This should be now 'view cookies' as suggested by the content designer.

File: src/components/cookies-banner/_macro.njk [hunk 5]
```diff
@@ -1,40 +1,84 @@
 {% macro onsCookiesBanner(params) %}
-    <div class="ons-cookies-banner" role="region" aria-label="{{ params.ariaLabel | default('Cookies banner')}}">
+    {% if params.lang == 'cy' %}
+        {% set ariaLabel = 'Cwcis' %}
+        {% set serviceName = 'ons.gov.uk' %}
+        {% set statementTitle = 'Cwcis ar' %}
+        {% set settingsLinkText = 'Gweld cwcis' %}
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro.njk:6]: This is 'view cookies', different to the current english 'cookie settings'.

File: src/components/cookies-banner/_macro.njk [hunk 6]
```diff
@@ -1,40 +1,84 @@
...
+        {% set preferencesText =  'Gallwch chi <a href="' + settingsLinkURL + '">newid eich dewisiadau o ran cwcis</a> ar unrhyw adeg.' %}
+        {% set confirmationButtonText = 'Cuddio' %}
+        {% set confirmationButtonTextAria = 'neges hon' %}
+    {% else %}
+        {% set ariaLabel = 'Cookies banner' %}
+        {% set serviceName = 'ons.gov.uk' %}
+        {% set statementTitle = 'Cookies on' %}
+        {% set settingsLinkText = 'Cookie settings' %}
+        {% set settingsLinkURL = '/cookies' %}
+        {% set statementText = '<p>Cookies are small files stored on your device when you visit a website. We store some cookies that are essential to make the site work.</p><p>We would like to set <a href="' + settingsLinkURL + '">additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services. </p>' %}
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro.njk:22]:
```suggestion
        {% set statementText = '<p>Cookies are small files stored on your device when you visit a website. We use some essential cookies to make this website work.</p><p>We would like to set <a href="' + settingsLinkURL + '">additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services. </p>' %}
```

File: src/components/cookies-banner/_macro.spec.js [hunk 7]
```diff
@@ -5,167 +5,225 @@ import * as cheerio from 'cheerio';
...
+        const $ = cheerio.load(renderComponent('cookies-banner', {}));
 
-    it('has provided `secondaryButtonUrl`', () => {
-      const faker = templateFaker();
-      const buttonSpy = faker.spy('button');
+        const statementText = $('.ons-cookies-banner__primary .ons-cookies-banner__statement')
+          .html()
+          .trim();
+        expect(statementText).toBe(
+          '<p>Cookies are small files stored on your device when you visit a website. We store some cookies that are essential to make the site work.</p><p>We would like to set <a href="/cookies">additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services. </p>',
```
Comment (jrbarnes9) [inline src/components/cookies-banner/_macro.spec.js:150]:
```suggestion
          '<p>Cookies are small files stored on your device when you visit a website. We store some cookies that are essential to make this website work.</p><p>We would like to set <a href="/cookies">additional cookies</a> to remember your settings and understand how you use the site. This helps us to improve our services.</p>',
```

File: src/patterns/help-users-to/cookies-settings/examples/cookie-settings-page/index.njk [hunk 8]
```diff
@@ -45,10 +45,12 @@ height="0" width="0" style="display:none;visibility:hidden"></iframe></noscript>
 {% from "components/cookies-banner/_macro.njk" import onsCookiesBanner %}
 {{
     onsCookiesBanner({
-        "statementTitle": 'Tell us whether you accept cookies',
-        "statementText": 'We use <a href="#0">cookies to collect information</a> about how you use census.gov.uk. We use this information to make the website work as well as possible and improve our services.',
-        "confirmationText": 'You’ve accepted all cookies. You can <a href="#0">change your cookie preferences</a> at any time.',
-        "secondaryButtonUrl": '#'
+        "statementTitle": 'Cookies on ons.gov.uk',
```
Comment (jrbarnes9) [inline src/patterns/help-users-to/cookies-settings/examples/cookie-settings-page/index.njk:48]: Service name should be separate param.
```suggestion
        "statementTitle": 'Cookies on',
```
Duplication in example:
Screenshot 2022-07-25 at 12 18 19

**Review by jrbarnes9 — APPROVED** [review]

**Review by kruncher — APPROVED** [review]

**Pull request merged**